### PR TITLE
SDK-1168. Update sync tests

### PR DESCRIPTION
**Pull request by aabellagm**
- Added a test to check enableSync after the remote node is removed and created again.

### CODE REVIEW

**Comment by aabellagm** [comment]
trigger compilation

File: tests/integration/SdkTest_test.cpp [hunk 1]
```diff
@@ -5137,6 +5371,477 @@ TEST_F(SdkTest, SyncResumptionAfterFetchNodes)
...
+
+        ASSERT_NO_FATAL_FAILURE(resumeSession(session.c_str()));
+        ASSERT_NO_FATAL_FAILURE(fetchnodes(0));
+        ASSERT_NO_FATAL_FAILURE(synchronousGetSpecificAccountDetails(0, true, false, false)); // Needed to ensure we know we are in OQ
+        sync = waitForSyncState(megaApi[0].get(), tagID, MegaSync::SYNC_DISABLED);
+        ASSERT_TRUE(sync && sync->getState() == MegaSync::SYNC_DISABLED);
+        ASSERT_EQ(MegaSync::STORAGE_OVERQUOTA, sync->getError());
+    }
+
+    LOG_verbose << "SyncOQTransitions :  Check for transition form OQ while offline.";
```
Comment (polmr) [inline tests/integration/SdkTest_test.cpp:5829]:
```suggestion
    LOG_verbose << "SyncOQTransitions :  Check for transition from OQ while offline.";
```

File: tests/integration/SdkTest_test.cpp [hunk 2]
```diff
@@ -5137,6 +5371,477 @@ TEST_F(SdkTest, SyncResumptionAfterFetchNodes)
...
+    ASSERT_NO_FATAL_FAILURE(locallogout());
+
+    std::unique_ptr<MegaNode> toRemoveNode(megaApi[1]->getChildNode(inshareNode, (remote1GBFile->getName() + to_string(filesNeeded-1)).c_str()));
+    ASSERT_EQ(API_OK, synchronousRemove(1, toRemoveNode.get()));
+
+    ASSERT_NO_FATAL_FAILURE(resumeSession(session.c_str()));
+    ASSERT_NO_FATAL_FAILURE(fetchnodes(0));
+    ASSERT_NO_FATAL_FAILURE(synchronousGetSpecificAccountDetails(0, true, false, false)); // Needed to ensure we know we are in OQ
```
Comment (polmr) [inline tests/integration/SdkTest_test.cpp:5837]:
```suggestion
    ASSERT_NO_FATAL_FAILURE(synchronousGetSpecificAccountDetails(0, true, false, false)); // Needed to ensure we know we are no longer in OQ
```

File: tests/integration/SdkTest_test.cpp [hunk 3]
```diff
@@ -803,7 +816,9 @@ int64_t SdkTest::getFilesize(string filename)
 
 void SdkTest::deleteFile(string filename)
 {
-    remove(filename.c_str());
+    fs::path p = fs::u8path(filename);
+    std::error_code ignoredEc;
+    fs::remove(p,ignoredEc);
```
[the remark this answers is not in the document]
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

File: tests/integration/SdkTest_test.cpp [hunk 4]
```diff
@@ -4913,10 +4968,192 @@ TEST_F(SdkTest, DISABLED_RecursiveDownloadWithLogout)
...
+    fs::path basePath1 = basePath / syncFolder1;
+    fs::path basePath2 = basePath / syncFolder2;
+    fs::path basePath3 = basePath / syncFolder3;
+    const auto localPath1 = fs::current_path() / basePath1;
+    const auto localPath2 = fs::current_path() / basePath2;
+    const auto localPath3 = fs::current_path() / basePath3;
+
+    ASSERT_NO_FATAL_FAILURE(cleanUp(this->megaApi[0].get(), basePath));
+
+    // Create local directory and a file.
```
[the remark this answers is not in the document]
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

File: tests/integration/SdkTest_test.cpp [hunk 5]
```diff
@@ -4913,10 +4968,192 @@ TEST_F(SdkTest, DISABLED_RecursiveDownloadWithLogout)
...
+    ASSERT_TRUE(sync && sync->getState() == MegaSync::SYNC_ACTIVE);
+    ASSERT_EQ(MegaSync::NO_SYNC_ERROR, sync->getError());
+    // Sync2
+    ASSERT_EQ(MegaError::API_OK, synchronousSyncFolder(0, localPath2.u8string().c_str(), remoteBaseNode2.get())) << "API Error adding a new sync";
+    ASSERT_EQ(MegaSync::NO_SYNC_ERROR, mApi[0].lastSyncError);
+    std::unique_ptr<MegaSync> sync2 = waitForSyncState(megaApi[0].get(), remoteBaseNode2.get(), MegaSync::SYNC_ACTIVE);
+    ASSERT_TRUE(sync2 && sync2->getState() == MegaSync::SYNC_ACTIVE);
+    ASSERT_EQ(MegaSync::NO_SYNC_ERROR, sync->getError());
+
+    LOG_verbose << "SyncRemoveRemoteNode :  Add syncs that fails";
```
[the remark this answers is not in the document]
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

File: tests/integration/SdkTest_test.cpp [hunk 6]
```diff
@@ -4913,10 +4968,192 @@ TEST_F(SdkTest, DISABLED_RecursiveDownloadWithLogout)
...
+    ASSERT_EQ(MegaSync::ACTIVE_SYNC_BELOW_PATH, mApi[0].lastSyncError);
+
+    LOG_verbose << "SyncRemoveRemoteNode :  Remove Syncs";
+    // Sync 1
+    ASSERT_EQ(MegaError::API_OK, synchronousRemoveSync(0, syncTag)) << "API Error removing the sync";
+    sync.reset(megaApi[0]->getSyncByNode(remoteBaseNode1.get()));
+    ASSERT_EQ(nullptr, sync.get());
+    // Sync 2
+    ASSERT_EQ(MegaError::API_OK, synchronousRemoveSync(0, sync2.get())) << "API Error removing the sync";
+    // Keep a not updated MegaSync object
```
[the remark this answers is not in the document]
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: The comment has been extended for further explanation

File: tests/integration/SdkTest_test.cpp [hunk 7]
```diff
@@ -4913,10 +4968,192 @@ TEST_F(SdkTest, DISABLED_RecursiveDownloadWithLogout)
...
+
+void cleanUp(::mega::MegaApi* megaApi, const fs::path &basePath)
+{
+
+    std::unique_ptr<MegaSyncList> syncs{megaApi->getSyncs()};
+    for (int i = 0; i < syncs->size(); i++)
+    {
+        RequestTracker removeTracker(megaApi);
+        megaApi->removeSync(syncs->get(i),&removeTracker);
+        removeTracker.waitForResult();
```
[the remark this answers is not in the document]
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

File: tests/integration/SdkTest_test.cpp [hunk 8]
```diff
@@ -5137,6 +5360,446 @@ TEST_F(SdkTest, SyncResumptionAfterFetchNodes)
...
+    megaApi[0]->setKeepSyncsAfterLogout(true);
+    ASSERT_NO_FATAL_FAILURE(logout(0));
+    auto trackerLogin = asyncRequestLogin(0, mApi[0].email.c_str(), mApi[0].pwd.c_str());
+    ASSERT_EQ(API_OK, trackerLogin->waitForResult()) << " Failed to establish a login/session for accout " << 0;
+    ASSERT_NO_FATAL_FAILURE(fetchnodes(0));
+    sync = waitForSyncState(megaApi[0].get(), tagID, MegaSync::SYNC_ACTIVE);
+    ASSERT_TRUE(sync && sync->getState() == MegaSync::SYNC_ACTIVE);
+    ASSERT_EQ(remoteFolder, string(sync->getMegaFolder()));
+
+    // Check if a logout with setKeepSyncsAfterLogout(false) doesn't keeps the sync configured.
```
[the remark this answers is not in the document]
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

File: tests/integration/SdkTest_test.cpp (hunk 1 above)
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

File: tests/integration/SdkTest_test.cpp (hunk 2 above)
Comment (aabellagm) [reply tests/integration/SdkTest_test.cpp]: Done

**Review by polmr — APPROVED** [review]

**Comment by polmr** [comment]
trigger compilation

**Pull request merged**